### PR TITLE
String examples

### PR DESCRIPTION
**Pull request by rjmunro**
Example string functions based on the regression tests to generate tests for.

### CODE REVIEW

File: diffblue.yml [hunk 1]
```diff
@@ -3,3 +3,5 @@ testCmd: cd main-module; mvn test
 cbmcArguments:
   java-max-input-array-length: 10
   unwind: 10
+  refine-strings: true
```
Comment (thisjunewest) [inline diffblue.yml:6]: I suggest we don't add this here, but wait for platform to be updated with it instead

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
+package com.diffblue.javatest;
+
+import java.io.PrintStream;
+import java.util.Locale;
+
+public class StringStuff {
```
Comment (thisjunewest) [inline main-module/src/main/java/com/diffblue/javatest/StringStuff.java:6]: How about "Strings", let's not slip back into rubbish naming

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+  public static String classGetNameExample(Class c) {
```
Comment (thisjunewest) [inline main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: I think "Example" is redundant in the method name, this entire repository is examples

Comment (thisjunewest) [inline main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: Can we avoid non-descriptive and one letter variable names? If people are using this a lot it should be easy to follow.

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+      return 0;
+
+    if (s.startsWith("</html>", q)) {
+      assert (false);
+      return 1;
+    } else{
+      return 0;}
+  }
+
+  public static int checkHtml2(String s) {
```
Comment (thisjunewest) [inline main-module/src/main/java/com/diffblue/javatest/StringStuff.java:332]: Can this be named to more clearly distinguish between `checkHtml`?

**Review by thisjunewest — CHANGES_REQUESTED** [review]
Please lint this (see README.md)

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 3 above)
Comment (rjmunro) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: @peterjwest The problem is that the names then become very close or identical to the things they are using.

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 2 above)
Comment (rjmunro) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:6]: Strings is rubbish as well - StringExamples or something would be better.

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 3 above)
Comment (rjmunro) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: The method bodies all come directly from the regression tests on CBMC - I'd rather leave them as is.
https://github.com/diffblue/test-gen/tree/master/regression/test-java-strings

Comment (thisjunewest) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: Is that a problem?

Comment (thisjunewest) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: I disagree, we're going to have to be looking through the trace debugging these, "s" and "c" aren't helpful.

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 4 above)
Comment (rjmunro) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:332]: I've removed it as it was hard to find in the original regression tests.

**Comment by rjmunro** [comment]
The linter wants me to add javadoc for every function, and make other changes that make the function bodies different to the originals. I'm not sure.

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 2 above)
Comment (royletron) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:6]: I'm with @rjmunro

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 3 above)
Comment (royletron) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: This time I am with @peterjwest how incredibly fair of me....

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 2 above)
Comment (thisjunewest) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:6]: StringExamples is ok

**Comment by sara-db** [comment]
@forejtv please review

File: diffblue.yml (hunk 1 above)
Comment (forejtv) [reply diffblue.yml:6]: @rjmunro has this already happened and can the line be removed?

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 2 above)
Comment (forejtv) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:6]: Yes, let's go with StringExamples

File: main-module/src/main/java/com/diffblue/javatest/StringStuff.java (hunk 3 above)
Comment (forejtv) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: I'm happy to keep `Example` as part of the name, so this is fine by me.

Comment (forejtv) [reply main-module/src/main/java/com/diffblue/javatest/StringStuff.java:8]: I'd say let's leave it as it is for now. If this becomes a problem let's modify it.

**Review by forejtv — APPROVED** [review]
See my comments, ow looks good.

**Comment by thisjunewest** [comment]
Closing as superseded by #18

**Pull request closed without merging**